src/components/satellitetracker/Satellite1.java

```diff
@@ -1,138 +1,204 @@
+     */
+
+    /*
+     * Added these constants to help with calculation of orbital speed and
      * period
      */
     private static final double PI = 3.14159265;
     private static final double G = 6.67430e-11; //
     private static final double M = 5.972e24;
+    private static final double GRAVITY = 9.81;
+    private static final double EARTH_RADIUS_KM = 6371.0;
+    private static final double MAX_ORBIT_RADIUS_KM = 384400.0;
 
-    public Satellite1(double x, double y, double z, double vx, double vy,
-            double vz) {
+    /**
+     * Representation of {@code this}.
+     */
+
+    private double[] position; // 3D Cartesian Coordinates
+    private double[] velocity; // Velocity components
 
-        // This is the constructor for SatelliteTracker object
-        this.x = x;
-        this.y = y;
-        this.z = z;
-        this.vx = vx;
-        this.vy = vy;
-        this.vz = vz;
+    /**
+     * Creator of initial representation (initialized to 0).
+     */
+    private void createNewRep() {
+        this.position = new double[] { 0.0, 0.0, 0.0 };
+        this.velocity = new double[] { 0.0, 0.0, 0.0 };
     }
 
-    public final void updatePosition(double time) {
-        // Final Position = Initial position + (Velocity in that direction)*(time)
-        this.x += this.vx * time;
-        this.y += this.vy * time;
-        this.z += this.vz * time;
+    /**
+     * Creator of initial representation (initialized to some user-prefered
+     * double values).
+     */
+    /**
+     * @param x
+     *            initial x coordinate
+     *
+     * @param y
+     *            initial y coordinate
+     *
+     * @param z
+     *            initial z coordinate
+     *
+     * @param vx
+     *            initial velocity in x direction
+     *
+     * @param vy
+     *            initial velocity in y direction
+     *
+     * @param vz
+     *            initial velocity in z direction
+     */
+    private void createNewRep(double x, double y, double z, double vx,
+            double vy, double vz) {
+        this.position = new double[] { x, y, z };
+        this.velocity = new double[] { vx, vy, vz };
     }
 
-    public final double[] getLiveLocation() {
+    /*
+     * Constructors -----------------------------------------------------------
+     */
 
-        // Returns live position in terms of x, y, z
-        /**
-         * Can later modify in the context of actual units used to describe
-         * position in space, if such units exist
-         */
-        return new double[] { this.x, this.y, this.z };
+    /**
+     * No-argument constructor.
+     */
+    public Satellite1() {
+        this.createNewRep();
     }
 
-    public final double[] getVelocity() {
-        return new double[] { this.vx, this.vy, this.vz };
+    /**
+     * Constructor from double (with parameters for position and velocity).
+     */
+    public Satellite1(double x, double y, double z, double vx, double vy,
```

Comment:
I don't think it really matters, but I might just make vector/point style record for use here:
```java
public record Vector(double x, double y, double z) {}
```